### PR TITLE
Update OP2Utility

### PR DESCRIPTION
**Pull request by Brett208**
Integrates new functions for filesystem management

### CODE REVIEW

File: src/ConsoleCreate.cpp [hunk 1]
```diff
@@ -73,14 +73,12 @@ vector<string> ConsoleCreate::GatherFilesForArchive(const vector<string>& paths)
...
 		if (XFile::IsDirectory(paths[i]))
 		{
-			vector<string> dirFilenames = XFile::GetFilenamesFromDirectory(paths[i]);
+			auto dirFilenames = XFile::DirFiles(paths[i]);
 
 			for (auto& filename : dirFilenames) {
 				filename = XFile::Append(paths[i], filename);
 			}
 
-			XFile::EraseNonFilenames(dirFilenames);
```
Comment (Brett208) [inline src/ConsoleCreate.cpp]: I believe this is no longer required with the new function DirFiles

File: src/ConsoleHelper.cpp [hunk 2]
```diff
@@ -34,13 +34,11 @@ unique_ptr<ArchiveFile> ConsoleHelper::OpenArchive(const string& archivePath)
 
 vector<string> ConsoleHelper::GetArchiveFilenames(const string& directory)
 {
-	vector<string> archiveFilenames = XFile::GetFilenamesFromDirectory(directory, ".vol");
-	vector<string> clmFilenames = XFile::GetFilenamesFromDirectory(directory, ".clm");
+	vector<string> archiveFilenames = XFile::DirFilesWithExtension(directory, ".vol");
+	vector<string> clmFilenames = XFile::DirFilesWithExtension(directory, ".clm");
 
 	archiveFilenames.insert(std::end(archiveFilenames), std::begin(clmFilenames), std::end(clmFilenames));
```
Comment (Brett208) [inline src/ConsoleHelper.cpp]: Lines 37-40 could be combined by using a regex call to DirFiles. This solution avoids the regex syntax, but requires combining two vectors. Just wanted to point it out.

Comment (DanRStevens) [reply src/ConsoleHelper.cpp]: Ahh, interesting observation. That might be worth doing. The regex approach is probably more efficient than combining the vectors after the fact.

File: src/ConsoleCreate.cpp (hunk 1 above)
Comment (DanRStevens) [reply src/ConsoleCreate.cpp]: Agreed, `DirFiles` removes the need to later call `EraseNonFilenames`.

**Pull request merged**